scripting/get5/pausing.sp

```diff
@@ -1,8 +1,17 @@
 public bool Pauseable() {
-  return g_GameState >= Get5State_KnifeRound && g_PausingEnabledCvar.BoolValue;
+  return g_GameState == Get5State_Live && g_PausingEnabledCvar.BoolValue;
```

Comment:
> Game can now only be paused when live. If you want a pause during knife, you must wait until live freezetime. I'm not 100% sure we want this, but it seems like calling a pause right after you readied your team is stupid. Admin can always pause.

Things can change in the 20 seconds between the match officially begins and the knife freeze time. I would rather this change not be made.

Reply:
I think it's also fine to let this go back to >=knife. I originally made this change to reduce logical complexity when refactoring, but it's been cleaned up so much that I don't see a problem with this. I'll just have to test it once.

Reply:
I think one other test that should be done is to check the warmup countdown right after the knife decision, as that reduces the amount of fixed pauses a team has per half, as reported here, without pausing the game - https://github.com/splewis/get5/issues/647 so maybe test this as well, just in case this still happens then? I think with this refactoring, that may no longer be an issue?

Reply:
The problem is that the game is restarted when the knife round ends, which I think unpauses? Will have to test.

It might actually be a bit of a problem to work around if that is in fact the case. @splewis is it a big deal to just let people pause in the freeze time after going live? They're going to end up paused at the same stage anyway, they just can't do it **while** still knifing for sides.

Reply:
Oh, the problem is of course if you want to pause in the **knife** freezetime? Because yea, we wouldn't allow that currently.

Reply:
Might be solvable by changing the if to `if (g_GameState >= Knife && g_GameState != WaitingForKnifeDecision)`.

Reply:
@splewis Fixed. You can pause during live, knife and waiting for knife decision and going live. I tested for https://github.com/splewis/get5/issues/647 and it works in all cases I can think of.